### ISSUE TITLE
Status not updating for previously declined cards.

### ISSUE THREAD

**Issue by danielsiemers** [solspace/craft3-freeform#1625]
### What happened?

I'm experiencing some issues with the stripe webhooks.

When a card is declined and then updated to a card that succeeds, I get a 500 error on the webhook when posting the payment_intent.successful event.

The error occurs in the StripeCallbackService.php file on like 89:
$metadata['type'] = $paymentMethod->type;

It says cannot read attribute type on a string.
This means that this piece of code above:

$paymentIntent = $stripe->paymentIntents->retrieve(
                $paymentIntent->id,
                ['expand' => ['customer', 'payment_method', 'invoice.subscription.plan.product']]
            );

Is not actually expanding the payment_method and instead returning the string for the payment id.

Any help here is appreciated, many thanks!

### Errors and Stack Trace (if available)

```shell
yii\base\ErrorException: Attempt to read property &quot;type&quot; on string in /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/solspace/craft-freeform/packages/plugin/src/Integrations/PaymentGateways/Stripe/Services/StripeCallbackService.php:89


Stack trace:
#0 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/craftcms/cms/src/web/ErrorHandler.php(79): yii\base\ErrorHandler-&gt;handleError(2, &apos;Attempt to read...&apos;, &apos;/container/appl...&apos;, 89)
#1 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/solspace/craft-freeform/packages/plugin/src/Integrations/PaymentGateways/Stripe/Services/StripeCallbackService.php(89): craft\web\ErrorHandler-&gt;handleError(2, &apos;Attempt to read...&apos;, &apos;/container/appl...&apos;, 89)
#2 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/solspace/craft-freeform/packages/plugin/src/Integrations/PaymentGateways/Stripe/Controllers/StripeWebhookController.php(74): Solspace\Freeform\Integrations\PaymentGateways\Stripe\Services\StripeCallbackService-&gt;handleSavedForm(Object(Solspace\Freeform\Form\Types\Regular), Object(Solspace\Freeform\Integrations\PaymentGateways\Stripe\Stripe), Object(Solspace\Freeform\Integrations\PaymentGateways\Stripe\Fields\StripeField), Object(Stripe\PaymentIntent), Object(Solspace\Freeform\Records\SavedFormRecord))
#3 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/solspace/craft-freeform/packages/plugin/src/Integrations/PaymentGateways/Stripe/Controllers/StripeWebhookController.php(51): Solspace\Freeform\Integrations\PaymentGateways\Stripe\Controllers\StripeWebhookController-&gt;handlePaymentIntent(Object(Stripe\Event))
#4 [internal function]: Solspace\Freeform\Integrations\PaymentGateways\Stripe\Controllers\StripeWebhookController-&gt;actionWebhooks()
#5 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#6 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction-&gt;runWithParams(Array)
#7 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/solspace/craft-freeform/packages/plugin/src/controllers/BaseApiController.php(15): yii\base\Controller-&gt;runAction(&apos;webhooks&apos;, Array)
#8 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/yiisoft/yii2/base/Module.php(552): Solspace\Freeform\controllers\BaseApiController-&gt;runAction(&apos;webhooks&apos;, Array)
#9 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/craftcms/cms/src/web/Application.php(349): yii\base\Module-&gt;runAction(&apos;freeform/stripe...&apos;, Array)
#10 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application-&gt;runAction(&apos;freeform/stripe...&apos;, Array)
#11 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/craftcms/cms/src/web/Application.php(317): yii\web\Application-&gt;handleRequest(Object(craft\web\Request))
#12 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application-&gt;handleRequest(Object(craft\web\Request))
#13 /container/application/releases/15b166078606ce95bcf2d0438a0e0bcd071fa5d1/public/index.php(12): yii\base\Application-&gt;run()
```

### How can we reproduce this?

1. Use a test card with an elevated risk 4000000000009235
2. Make a test payment on a stripe field
3. Ensure the card is decline at first and then authorised on a subsequent request
4. Check the submission to see the status for the payment is "
Requires payment method"

### Freeform Edition

Pro

### Freeform Version

5.6.8

### Craft Version

5.3.6

### When did this issue start?

Fresh install of Freeform

**Comment by kjmartens** [comment]
Sorry for the delay and the trouble you're experiencing @danielsiemers.

We will investigate this issue shortly. 🙂

**Comment by kjmartens** [comment]
Hi @danielsiemers,

This will be resolved in Freeform 5.7.3, which will be available later today. 🙂

**Comment by danielsiemers** [comment]
Amazing! 
Will give it a go when this releases.

Thank you @kjmartens!
